### ISSUE TITLE
BigQuery Connection Issue From Hasura CLI Console

### ISSUE THREAD

**Issue by ImElan** [hasura/graphql-engine#9172]
### Version Information

- Server Version: **v2.13.0.cli-migrations-v3**
- CLI Version (for CLI related issue): **v2.14.0-beta.2**

### Environment
- Open Source Software

### What is the current behaviour?
When i'm trying to connect to a Big Query Database from Hasura Console spun up by Hasura CLI, i'm getting the following error **"cannot determine source kind for <my_database_name>"** as a console alert.

I've further explored to see which network call is failing, as it turns out

`{{CLI_CONSOLE_HOST_URL}}/apis/migrate`, this API Call is failing with a `500 Internal Server Error` and the error response is as follows,

```
{
    "code": "request_parse_error",
    "message": "cannot determine source kind for <my_database_name>"
}
```

This only happens from CLI Console and not Server Console, from Server console i'm able to establish a connection to my Big Query Database with the same configurations. So i'm guessing something is going wrong when the CLI Console is trying to parse my configs and write it to the configuration files.

### What is the expected behaviour?
- A Connection to the Big Query Database should be established.
- The Metadata and Migrations directory should be updated with the configuration details about the connection.

### How to reproduce the issue?

1. Spin up a **graphql-engine** with the above mentioned version as a docker container.
2. Install the above mentioned Hasura CLI tool.
3. Once the **graphql-engine** is up, type `hasura console` in the terminal to spin up the CLI console.
4. Navigate to the CLI console (usually runs in localhost:9695).
5. Open the data tab from the navigation bar and click on connect database.
6. Select BigQuery from the dropdown and provide the necessary details like datasets, database name, service-account.json file (exported it as an environment variable and give the environment variable name).
7. After filling in all the details click on connect.
8. Now the error alert with the above mentioned error message should appear in the top right of the browser.

### Please provide any traces or logs that could help here.
There are no logs for this, But as mentioned before I can give the failed API call details

**Request URL**
```
POST -> {{CLI_CONSOLE_HOST_URL}}/apis/migrate
```

**Payload**
```
```

**Response**
```
{
    "code": "request_parse_error",
    "message": "cannot determine source kind for <my_database_name>"
}
```

### Any possible solutions/workarounds you're aware of?

- The main reason I want to use the console spun up by CLI instead of the server console is, to get the configurations automatically getting reflected in my metadata and migrations directory. Because of the above mentioned issue I'm missing this functionality.
- So the workaround what I did was to add the connections from Server Console (localhost:8080) and then used the command `hasura metadata export` to manually load the configurations inside the metadata directory. This solves the metadata problem.
- And for the migrations, during automatic apply process because of the CLI Base Image during the hasura engine startup, it expects a folder with the name of the database which got added. So I manually created that folder under migrations directory with the same name as my Big Query Database.

### Keywords
- adding data source failed
- cli console big query connection issue
- request parse error
- cannot determine source kind

**Comment by abhi40308** [comment]
I think bigquery is not recognised by CLI currently, we need to add it to the types https://pkg.go.dev/github.com/hasura/graphql-engine/cli/v2/internal/hasura#SourceKind

**Comment by manasag** [comment]
We have now added support for BigQuery in CLI Console. Please checkout v2.20.0-beta.1

**Comment by ImElan** [comment]
@manasag Yes it works now, I'm able to establish a BigQuery connection from Hasura CLI and the metadata & migrations directory in my file structure is getting reflected with the changes. Thanks
